db/migrate/20230515141021_add_summary_and_body_to_worldwide_organisation_translations.rb

```diff
@@ -0,0 +1,48 @@
+class AddSummaryAndBodyToWorldwideOrganisationTranslations < ActiveRecord::Migration[7.0]
+  class Edition < ApplicationRecord
+  end
+
+  class CorporateInformationPage < Edition
+    ABOUT_US_TYPE_ID = 20
+    scope :published, -> { where(state: "published") }
+    translates :summary, :body
+  end
+
+  class WorldwideOrganisation < ApplicationRecord
+    has_many :edition_worldwide_organisations
+    has_many :corporate_information_pages, through: :edition_worldwide_organisations, source: :edition, class_name: "::CorporateInformationPage"
+    translates :summary, :body
+
+    def about_us
+      @about_us ||= corporate_information_pages.published.find_by(
+        corporate_information_page_type_id: CorporateInformationPage::ABOUT_US_TYPE_ID,
+      )
+    end
+  end
+
+  def up
+    change_table :worldwide_organisation_translations, bulk: true do |t|
+      t.column :summary, :text
+      t.column :body, :text
+    end
+
+    WorldwideOrganisation.all.each do |worldwide_organisation|
+      next if worldwide_organisation.about_us.nil?
+
+      worldwide_organisation.about_us.translations.each do |source_translation|
+        target_translation = worldwide_organisation.translations.find_by(locale: source_translation.locale)
+        if target_translation.nil?
+          target_translation = worldwide_organisation.translations.create!(locale: source_translation.locale)
```

Comment:
I guess this has the impact of publishing translations for worldwide organisations that did not previously have them. I had a look at a few translations in integration that fell into this camp, and they seem to have been created this year in March (i.e. they're not so old that nobody cares about them, but old enough that I'm surprised they haven't been published or complained about). I'm wondering if we should add these to the list of things we want to check with FDCO that they're OK with being published before we merge this.

Reply:
We've added this to our list of things to ask FCDO about. Marking as resolved.